jerry-core/ecma/base/ecma-helpers.h

```diff
@@ -127,13 +127,19 @@ typedef enum
  * Set an internal property value from pointer.
  */
 #define ECMA_SET_INTERNAL_VALUE_POINTER(field, pointer) \
-  (field) = ((ecma_value_t) pointer)
+  ((field) = ((ecma_value_t) pointer))
 
 /**
  * Set an internal property value from pointer. Pointer can be NULL.
  */
 #define ECMA_SET_INTERNAL_VALUE_ANY_POINTER(field, pointer) \
-  (field) = ((ecma_value_t) pointer)
+  ((field) = ((ecma_value_t) pointer))
+
+/**
+ * Set an internal property to NULL.
+ */
+#define ECMA_SET_INTERNAL_VALUE_TO_NULL(field) \
```

Comment:
Do we really need to add new macros for this? Can't we just use `ECMA_SET_INTERNAL_VALUE_ANY_POINTER`?

Reply:
Inefficient, and the code does not look good. Eg.:
```
while (IS_NOT_NULL(something)) {
  something = DECODE(something)->next
}
```
Looks like this:
```
while (true) {
  something_p = DECODE_ANY(something)
  // NULL checked twice
  if (something_p == NULL) {
   break;
  }
  something = something_p->next
}
```

Reply:
The question was regarding `ECMA_SET_INTERNAL_VALUE_TO_NULL`, not `ECMA_IS_INTERNAL_VALUE_NULL`.

Reply:
As a matter of fact, couldn't we use compressed pointers directly instead of internal values? Do we need tag flags for something?

Reply:
When compression does not needed pointers can be stored directly, so these macros are preferrable (they were invented for this purpose).

I am sorry I misunderstand your question. `ECMA_SET_INTERNAL_VALUE_ANY_POINTER` could be used, although I think these are just macros, and they should not hurt, but I can remove them if you want.

Reply:
I would prefer to not add specialized macros when we can use a more general one we already have.